src/runtime_src/hip/core/device.h

```diff
@@ -3,12 +3,51 @@
 #ifndef xrthip_device_h
 #define xrthip_device_h
 
+#include "core/common/api/handle.h"
+#include "xrt/xrt_device.h"
+
+#include <vector>
+
 namespace xrt::core::hip {
+/**
+ * typedef device_handle - opaque device handle
+ */
+typedef uint32_t device_handle;
+
+// forward declaration
+class context;
 
 class device
 {
+  xrt::device m_xrt_device;
+  uint32_t m_device_id = UINT32_MAX;
```

Comment:
Looks like this needs <cstdint> to be included.
It is custom to do default initialization as `var {value}`  that that it makes a difference.